Update image file paths for terraform-docs-agents

heatlikeheatwave opened the pull request:
[Asana task](https://app.asana.com/1/90955849329269/project/1209399665311208/task/1210006836405224?focus=true)

## Changes
- Update image paths so they do not 404 in production. The `/public/img` path is an edge-case that has been corrected in this PR (see typical path [here](https://github.com/hashicorp/terraform-docs-agents?tab=readme-ov-file#content-images))
- Add a default image for themed images (e.g. `-dark.png`)
- Add TODOs to uncomment and remove default images when themed images are supported in UDR.

Note: terraform-docs-agents docs do not currently support themed images (see [this Slack thread](https://hashicorp.slack.com/archives/C5FSPUGDS/p1734030591207049?thread_ts=1723843460.759869&cid=C5FSPUGDS)); themed images for these docs would not be feature parity with the current API, it would be new functionality, making it out of scope for this PR.

## Testing
These endpoints should return markdown with two `TODO` notes, like this:

### Request forwarding
```
<>{/* TODO: uncomment these when themed images are supported in UDR ![Request forwarding architecture diagram](/img/request-forwarding-arch-light.png#light-theme-only) ![Request forwarding architecture diagram](/img/request-forwarding-arch-dark.png#dark-theme-only) */}</>\n<>{/* TODO: delete the image below when themed images are supported in UDR */}</>\n![Request forwarding architecture diagram](/img/request-forwarding-arch-dark.png)
```

https://web-unified-docs-oqgekzemi-hashicorp.vercel.app/api/content/terraform-docs-agents/doc/v1.15.x/cloud-docs/agents/request-forwarding
https://web-unified-docs-oqgekzemi-hashicorp.vercel.app/api/content/terraform-docs-agents/doc/v1.16.x/cloud-docs/agents/request-forwarding
https://web-unified-docs-oqgekzemi-hashicorp.vercel.app/api/content/terraform-docs-agents/doc/v1.17.x/cloud-docs/agents/request-forwarding
https://web-unified-docs-oqgekzemi-hashicorp.vercel.app/api/content/terraform-docs-agents/doc/v1.18.x/cloud-docs/agents/request-forwarding
https://web-unified-docs-oqgekzemi-hashicorp.vercel.app/api/content/terraform-docs-agents/doc/v1.19.x/cloud-docs/agents/request-forwarding
https://web-unified-docs-oqgekzemi-hashicorp.vercel.app/api/content/terraform-docs-agents/doc/v1.20.x/cloud-docs/agents/request-forwarding
https://web-unified-docs-oqgekzemi-hashicorp.vercel.app/api/content/terraform-docs-agents/doc/v1.21.x/cloud-docs/agents/request-forwarding
https://web-unified-docs-oqgekzemi-hashicorp.vercel.app/api/content/terraform-docs-agents/doc/v1.22.x/cloud-docs/agents/request-forwarding

### Telemetry

```
<>{/* TODO: uncomment these when themed images are supported in UDR ![telemetry architecture diagram](/img/telemetry-arch-light.png#light-theme-only) ![telemetry architecture diagram](/img/telemetry-arch-dark.png#dark-theme-only) */}</>\n<>{/* TODO: delete the image below when themed images are supported in UDR */}</>\n![telemetry architecture diagram](/img/telemetry-arch-light.png)
```

https://web-unified-docs-oqgekzemi-hashicorp.vercel.app/api/content/terraform-docs-agents/doc/v1.15.x/cloud-docs/agents/telemetry
https://web-unified-docs-oqgekzemi-hashicorp.vercel.app/api/content/terraform-docs-agents/doc/v1.16.x/cloud-docs/agents/telemetry
https://web-unified-docs-oqgekzemi-hashicorp.vercel.app/api/content/terraform-docs-agents/doc/v1.17.x/cloud-docs/agents/telemetry
https://web-unified-docs-oqgekzemi-hashicorp.vercel.app/api/content/terraform-docs-agents/doc/v1.18.x/cloud-docs/agents/telemetry
https://web-unified-docs-oqgekzemi-hashicorp.vercel.app/api/content/terraform-docs-agents/doc/v1.19.x/cloud-docs/agents/telemetry
https://web-unified-docs-oqgekzemi-hashicorp.vercel.app/api/content/terraform-docs-agents/doc/v1.20.x/cloud-docs/agents/telemetry
https://web-unified-docs-oqgekzemi-hashicorp.vercel.app/api/content/terraform-docs-agents/doc/v1.21.x/cloud-docs/agents/telemetry
https://web-unified-docs-oqgekzemi-hashicorp.vercel.app/api/content/terraform-docs-agents/doc/v1.22.x/cloud-docs/agents/telemetry

LeahMarieBush reviewed (commented):
The v1.15.x and v1.16.x versions of the images for both the telemetry page and the request forwarding page are not loading

Ex: https://unified-docs-frontend-preview-jztmb81f7-hashicorp.vercel.app/terraform/cloud-docs/agents/v1.15.x/request-forwarding

heatlikeheatwave commented:
> The v1.15.x and v1.16.x versions of the images for both the telemetry page and the request forwarding page are not loading
> 
> Ex: https://unified-docs-frontend-preview-jztmb81f7-hashicorp.vercel.app/terraform/cloud-docs/agents/v1.15.x/request-forwarding

fixed!

lafentres commented:
> > > The v1.15.x and v1.16.x versions of the images for both the telemetry page and the request forwarding page are not loading
> > > Ex: https://unified-docs-frontend-preview-jztmb81f7-hashicorp.vercel.app/terraform/cloud-docs/agents/v1.15.x/request-forwarding
> > 
> > 
> > fixed!
> 
> The images on these pages still aren't loading for me. The links you have in the description for the PR all look good to me but on the dev-portal preview link those specific images still aren't working. I also tried locally and it's also not working there

i checked the existing docs and it looks like these images don't load for v1.15 or v1.16 either. i don't know why that is the case but it seems like we may have accidentally broken something a while back for those versions

LeahMarieBush reviewed (approved):
Since the images in prod for v1.15.x and v1.16.x also don't work I'm going to approve this. We should write a ticket to fix it after launching terraform-docs-agents. Thanks for the fix!

heatlikeheatwave commented:
> > > The v1.15.x and v1.16.x versions of the images for both the telemetry page and the request forwarding page are not loading
> > > Ex: https://unified-docs-frontend-preview-jztmb81f7-hashicorp.vercel.app/terraform/cloud-docs/agents/v1.15.x/request-forwarding
> > 
> > 
> > fixed!
> 
> The images on these pages still aren't loading for me. The links you have in the description for the PR all look good to me but on the dev-portal preview link those specific images still aren't working. I also tried locally and it's also not working there

I am extremely confused. I deleted and added the images from another version which does work, double-checked the file path, and it _still_ will not load. Not really sure what to do here. Since it's broken in production currently as @lafentres mentioned, maybe we can just leave it as is. Thought @LeahMarieBush?

LeahMarieBush commented:
> > > > The v1.15.x and v1.16.x versions of the images for both the telemetry page and the request forwarding page are not loading
> > > > Ex: https://unified-docs-frontend-preview-jztmb81f7-hashicorp.vercel.app/terraform/cloud-docs/agents/v1.15.x/request-forwarding
> > > 
> > > 
> > > fixed!
> > 
> > 
> > The images on these pages still aren't loading for me. The links you have in the description for the PR all look good to me but on the dev-portal preview link those specific images still aren't working. I also tried locally and it's also not working there
> 
> I am extremely confused. I deleted and added the images from another version which does work, double-checked the file path, and it _still_ will not load. Not really sure what to do here. Since it's broken in production currently as @lafentres mentioned, maybe we can just leave it as is. Thought @LeahMarieBush?

Yeah I agree we can leave it as is
